annotations/crd-annotations/src/main/java/io/dekorate/crd/apt/CustomResourceAnnotationProcessor.java

```diff
@@ -15,28 +15,48 @@
  */
 package io.dekorate.crd.apt;
 
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Stream;
 
 import javax.annotation.processing.RoundEnvironment;
 import javax.annotation.processing.SupportedAnnotationTypes;
 import javax.lang.model.element.Element;
 import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.MirroredTypeException;
 
 import io.dekorate.config.MultiConfiguration;
 import io.dekorate.crd.adapter.CustomResourceConfigAdapter;
-import io.dekorate.crd.annotation.CustomResource;
+import io.dekorate.crd.annotation.Crd;
 import io.dekorate.crd.config.Keys;
+import io.dekorate.crd.config.Scope;
 import io.dekorate.crd.config.CustomResourceConfig;
 import io.dekorate.crd.config.CustomResourceConfigBuilder;
 import io.dekorate.crd.configurator.AddClassNameConfigurator;
 import io.dekorate.crd.generator.CustomResourceGenerator;
+import io.dekorate.crd.util.Types;
 import io.dekorate.processor.AbstractAnnotationProcessor;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Version;
+import io.fabric8.kubernetes.model.annotation.Kind;
+import io.fabric8.kubernetes.model.annotation.Plural;
+import io.fabric8.kubernetes.model.annotation.Singular;
 import io.sundr.codegen.CodegenContext;
 import io.sundr.codegen.functions.ElementTo;
 import io.sundr.codegen.model.TypeDef;
 import io.sundr.codegen.utils.ModelUtils;
+import io.sundr.codegen.utils.TypeUtils;
 
-@SupportedAnnotationTypes({ "io.dekorate.crd.annotation.CustomResource" })
+@SupportedAnnotationTypes({
+    "io.fabric8.kubernetes.model.annotation.Group",
+    "io.fabric8.kubernetes.model.annotation.Version",
+    "io.fabric8.kubernetes.model.annotation.Kind",
+    "io.fabric8.kubernetes.model.annotation.Plural",
+    "io.fabric8.kubernetes.model.annotation.Singular",
+    "io.dekorate.crd.annotation.Crd"})
```

Comment:
out of curiosity, why we can use annotations from fabric8 but then we need a new one `Crd` in dekorate instead of using the `CustomResource` in fabric8?

Reply:
@ppatierno because the annotations in fabric8 don't cover all that's needed to generate a (more) complete CRD. See fabric8io/kubernetes-client#2736 and fabric8io/kubernetes-client#2737.

Reply:
So it sounds that contributing to fabric8 improving the `CustomResource` allows removing this `Crd` in the future, right?

Reply:
@ppatierno yes, that's the goal at this point